include/souper/Inst/Inst.h

```diff
@@ -119,6 +120,7 @@ struct Inst : llvm::FoldingSetNode {
   static int getCost(Kind K);
   llvm::APInt KnownZeros;
   llvm::APInt KnownOnes;
+  llvm::APInt DemandedBitsVal = llvm::APInt(1, 0, false);
```

Comment:
I am just initializing the demanded bits value to 0, otherwise it will take some garbage values from the places where I am not setting its value. Width of 1 is just for initialization purpose and it gets over-written with exact width and value wherever we get some concrete values to assign it to. Similarly, I did for known zeros/ones, the only difference is I was passing them as an argument to createVar() function and thus initialized the function arguments value to llvm::APInt(1, 0, false)